r/src/safe-call-into-r-impl.cpp

```diff
@@ -46,11 +46,7 @@ bool SetEnableSignalStopSource(bool enabled) {
 
 // [[arrow::export]]
 bool CanRunWithCapturedR() {
-#if defined(HAS_UNWIND_PROTECT)
   return MainRThread::GetInstance().Executor() == nullptr;
-#else
-  return false;
-#endif
```

Comment:
Do we need this function at all anymore? Or another way: does `MainRThread::GetInstance().Executor() == nullptr` ever return false?

Reply:
:shrug: I have no idea tbh. Just following the prompt from the cpp11 author ^^

Reply:
ah, the header explains it, I'll update the comment but from how I read it we still need to check for the executor
>// Unwind protection was added in R 3.5 and some calls here use it
// and crash R in older versions (ARROW-16201). Implementation provided
// in safe-call-into-r-impl.cpp so that we can skip some tests
// when this feature is not provided. This also checks that there
// is not already an event loop registered (via MainRThread::Executor()),
// because only one of these can exist at any given time.